index.bs

```diff
@@ -1402,6 +1407,100 @@ lt="ReadableStreamDefaultReader(stream)">new ReadableStreamDefaultReader(<var>st
+  1. Return _iterator_.
+</emu-alg>
+
+<!-- Bikeshed doesn't let us mark this up correctly: https://github.com/tabatkins/bikeshed/issues/1344 -->
+<h5 id="default-reader-@@asynciterator">[@@asyncIterator]()</h5>
+
+<p class="note">
+  The <code>@@asyncIterator</code> method is an alias of {{ReadableStreamDefaultReader/getIterator()}}.
+</p>
+
+The initial value of the <code>@@asyncIterator</code> method is the same function object as the initial value of the
+{{ReadableStreamDefaultReader/getIterator()}} method.
+
+<h3 id="default-reader-asynciterator-prototype" interface
+lt="ReadableStreamDefaultReaderAsyncIteratorPrototype">ReadableStreamDefaultReaderAsyncIteratorPrototype</h3>
+
+{{ReadableStreamDefaultReaderAsyncIteratorPrototype}} is an ordinary object that is used by
+{{ReadableStreamDefaultReader/getIterator()}} to construct the objects it returns. Instances of
+{{ReadableStreamDefaultReaderAsyncIteratorPrototype}} implement the {{AsyncIterator}} abstract interface from the
+JavaScript specification. [[!ECMASCRIPT]]
+
+The {{ReadableStreamDefaultReaderAsyncIteratorPrototype}} object must have its \[[Prototype]] internal slot set to
+{{%AsyncIteratorPrototype%}}.
+
+<p class="note">The pattern here of having a prototype without a corresponding class/constructor is unusual, but
+emulates similar practices in the JavaScript and Web IDL specifications, when they create iterators.</p>
+
+<h4 id="default-reader-asynciterator-prototype-internal-slots">Internal slots</h4>
+
+Objects created by {{ReadableStreamDefaultReader/getIterator()}}, using
+{{ReadableStreamDefaultReaderAsyncIteratorPrototype}} as their prototype, are created with the internal slots described
+in the following table:
+
+<table>
+  <thead>
+    <tr>
+      <th>Internal Slot</th>
+      <th>Description (<em>non-normative</em>)</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>\[[asyncIteratorReader]]
+    <td class="non-normative">A {{ReadableStreamDefaultReader}} instance
+  </tr>
+  <tr>
+    <td>\[[preventCancel]]
+    <td class="non-normative">A boolean value indicating if the stream will be <a lt="cancel a readable
+    stream">canceled</a> when the async iterator's {{ReadableStreamDefaultReaderAsyncIteratorPrototype/return()}} method
+    is called
+  </tr>
+</table>
+
+<h4 id="default-reader-asynciterator-prototype-next" method
+for="ReadableStreamDefaultReaderAsyncIteratorPrototype">next()</h4>
+
+<emu-alg>
+  1. If ! IsReadableStreamDefaultReaderAsyncIterator(*this*) is *false*, throw a *TypeError* exception.
+  1. Let _reader_ be *this*.[[asyncIteratorReader]].
+  1. Let _read_ be GetMethod(_reader_, `"read"`).
+  1. If _read_ is an abrupt completion, return <a>a promise rejected with</a> _read_.[[Value]].
+  1. Let _result_ be Call(_read_.[[Value]], _reader_).
+  1. If _result_ is an abrupt completion, return <a>a promise rejected with</a> _result_.[[Value]].
+  1. Return <a>a promise resolved with</a> _result_.[[Value]].
```

Comment:
This is missing auto-release if _result_.[[Value]]'s `done` property is set to true, discussed in https://github.com/whatwg/streams/issues/778#issuecomment-371717821